packages/components/src/internal/components/search/FilterFacetedSelector.tsx

```diff
@@ -1,64 +1,162 @@
-import React, { FC, memo, useEffect, useState } from 'react';
+import React, { FC, memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
 
-import { Query } from '@labkey/api';
+import { Filter, Query } from '@labkey/api';
 
 import { naturalSort } from '../../../public/sort';
 import { Alert } from '../base/Alert';
 import { resolveErrorMessage } from '../../util/messaging';
 import { LoadingSpinner } from '../base/LoadingSpinner';
 
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
+
+import { ALL_VALUE_DISPLAY, EMPTY_VALUE_DISPLAY, getCheckedFilterValues, getUpdatedChooseValuesFilter } from './utils';
+
 interface Props {
+    api?: ComponentsAPIWrapper;
     selectDistinctOptions: Query.SelectDistinctOptions;
+    fieldKey: string;
+    fieldFilter: Filter.IFilter;
+    onFieldFilterUpdate?: (newFilter: Filter.IFilter) => void;
+    showSearchLength?: number; // show search box if number of unique values > N
 }
 
 export const FilterFacetedSelector: FC<Props> = memo(props => {
-    const { selectDistinctOptions } = props;
+    const { api, selectDistinctOptions, fieldKey, fieldFilter, onFieldFilterUpdate, showSearchLength } = props;
 
-    const [activeFieldDistinctValues, setActiveFieldDistinctValues] = useState<any[]>(undefined);
+    const [fieldDistinctValues, setFieldDistinctValues] = useState<string[]>(undefined);
     const [error, setError] = useState<string>(undefined);
+    const [searchStr, setSearchStr] = useState<string>(undefined);
 
     useEffect(() => {
-        Query.selectDistinctRows({
-            ...selectDistinctOptions,
-            success: result => {
-                const distinctValues = result.values.sort(naturalSort);
-                setActiveFieldDistinctValues(distinctValues);
-            },
-            failure: error => {
+        api.query
+            .selectDistinctRows(selectDistinctOptions)
+            .then(result => {
+                const distinctValues = result.values
+                    .map(val => {
+                        if (val === '' || val === null || val === undefined) return EMPTY_VALUE_DISPLAY;
+                        return val;
+                    })
+                    .sort(naturalSort);
+                distinctValues.unshift(ALL_VALUE_DISPLAY);
+                setFieldDistinctValues(distinctValues);
+            })
+            .catch(error => {
                 console.error(error);
                 setError(resolveErrorMessage(error));
-            },
+            });
+    }, [selectDistinctOptions, fieldKey]);
```

Comment:
is `fieldKey` needed here in the deps array? I am not seeing it used in this `useEffect`?

Reply:
oh is it so that the new distinct values are loaded when the user selects a different field from the UI?

Reply:
yep. added comment to clarify